src/pages/Detail/components/VarietyOfPokemon/index.js

```diff
@@ -0,0 +1,53 @@
+import axios from 'axios'
+import { useCallback, useEffect, useState } from 'react'
+import { Link } from 'react-router-dom'
+
+const VarietyOfPokemon = ({ pokemon, index }) => {
+  // save state detail pokemon.
+  const [detailPokemon, setDetailPokemon] = useState(() => ({
+    id: '',
+    name: '',
+    front_default: ''
+  }))
+
+  // get detail of pokemon.
+  const fetchDetailPokemon = useCallback(url => {
+    const _url = `${url}`.replaceAll('-species', '')
+    return axios.get(_url).then(response => response.data)
+  }, [])
+
+  useEffect(() => {
+    fetchDetailPokemon(`${pokemon?.url}`).then(response => {
+      // assignment data.
+      let data = {}
+      data.id = response?.id
+      data.name = response?.name
+      data.front_default = response?.sprites?.front_default
+
+      // set data for state.
+      setDetailPokemon(data)
+    })
```

Comment:
what will happen if fetching pokemon details become failure?

Reply:
it will not show data, what the best way to fix this?

Reply:
we should introduce an error-state (or exception) UI for that case, to inform users what problem(s) occurred
For example: 

<img width="485" alt="Screenshot 2022-01-10 at 17 12 06" src="https://user-images.githubusercontent.com/52815696/148749148-eacf28fe-4163-43e3-8b3f-916438d17682.png">

Reply:
For any  error state, right?
i see the example above like 404 error, but this for some url or uri not found

Reply:
we can call that is general error UI

Reply:
404 error is another kind of error